docs/scopes_importation.md

```diff
@@ -0,0 +1,211 @@
+# Sign up through CSV
+
+In order to sign up users, we need a csv with format:
+
+```
+DNI,nom,Email,Tmobil,codi
+
+123456789A,Test User,user@example.org,650650650,2601
+```
+
+This fields are required for sign up. The last field, code, in this case the code 2601, belongs to a local scope. We don't know what code belongs to user and it is given to us in csv. This code must be the **local code** in hierarchy.
+
+## Required CSVs (files)
+
+To be able to generate code mapping, we need the following files:
+
+
+### **scopes_codes.csv**
+
+```
+code,name,type
+
+0002,Girona,FR
+```
+
+In this example, we have the code 0002 that belongs to Girona, which is a region.
+
+### **scopes_hierarchy.csv**
+
+```
+FR,FC,SL
+
+0002,5100,5117
+```
+In this example, we have relation between Girona's code and their local and comarcal codes. In this case, 5100 is the relate comarcal code and 5117 is the relate local code.
+
+Caption:
+
+**FR**: regional code
+**FC**: comarcal code
+**SL**: local code
+
+## Rake tasks for run
+
+Next step after have the files is execute rake tasks in order to generate required yml files.
+
+The tasks must be run in this order:
+```
+rake erc_auth:csv_import:comarcals
+rake erc_auth:csv_import:locals
+rake erc_auth:csv_import:regionals 
+rake erc_auth:csv_import:local_comarcal_relationships
+rake erc_auth:csv_import:local_regional_relationships 
+rake civi_crm:generate:comarcal_exceptions  
+rake civi_crm:generate:decidim_scopes_mapping
+rake civi_crm:create:scopes 
+```
+
+## 1. rake erc_auth:csv_import:comarcals
+
+This tasks generate the file `comarcals.yml` from `scopes_codes.csv`.
+
+Check the rows it has FC code and add to yml.
+
+`comarcals.yml` contain:
+
+```
+---
+'0100': Alacantí
+'0200': Alcalatén
+'0300': Alcoià
+'0400': Alt Camp
+'0500': Alt Empordà
+'0600': Alt Maestrat
+```
+
+## 2. rake erc_auth:csv_import:locals
+
+This task generate the file `locals.yml` from `scopes_codes.csv`.
+
+Check the rows it has SL code and add to yml.
+
+`locals.yml` contains:
+
+```
+---
+'0103': Alacant
+'0106': Mutxamel
+'0110': Xixona
+'0201': L'Alcora
+'0301': Alcoi
+'0402': Alcover
+```
+
+## 3. rake erc_auth:csv_import:regionals 
+
+This task generate the file `regionals.yml` from `scopes_codes.csv`.
+
+Check it the rows it has FR code and add to yml.
+
+`regionals.yml` contains:
+
+```
+---
+'0001': Regió Metropolitana de Barcelona
+'0002': Girona
+'0003': Camp de Tarragona
+'0004': Ebre
+```
+
+## 4. rake erc_auth:csv_import:local_comarcal_relationships
+
+This task generate the file `local_comarcal_relationships.yml` from the files `locals.yml`, `comarcals.yml` and `scopes_hierarchy.csv`.
+
+Search local code and in the same row, get FC (comarcal) code.
```

Comment:
So the task searches the local code (from which file?) in the same row (same as what?), then gets the comarcal code, and....
Can you please better explain? :smile: